example/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"github.com/Sirupsen/logrus"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/longhorn/upgrade-responder/client"
+)
+
+const (
+	// example.com should be replaced by your Upgrade Responder server address
+	upgradeResponderServerAddress = "https://example.com/v1/checkupgrade"
+
+	version           = "v0.8.1"
+	kubernetesVersion = "v1.19.1"
+)
+
+func main() {
+	done := make(chan struct{})
+
+	upgradeChecker := client.NewUpgradeChecker(upgradeResponderServerAddress, &MyUpgradeRequester{})
+	upgradeChecker.Start()
+	defer upgradeChecker.Stop()
+
+	registerShutdownChannel(done)
+	<-done
+}
+
+type MyUpgradeRequester struct{}
+
+func (f *MyUpgradeRequester) GetCurrentVersion() string {
+	return version
+}
+
+func (f *MyUpgradeRequester) GetExtraInfo() map[string]string {
+	return map[string]string{"kubernetesVersion": kubernetesVersion}
+}
+
+func (f *MyUpgradeRequester) ProcessUpgradeResponse(latestVersion string, err error) {
```

Comment:
We're returning an array of versions with different tags. What if the project has stable and latest versions?

Pickup the version should be handled by the client. They should know the tag since the authors are who set it.

Reply:
I updated the client to return the `CheckUpgradeResponse` to the user

I think we should leave user to handle the `err`, they may want to record the `err` in a certain format

I still need to modify the server to return all tags. Right now, it only returns the latest tag https://github.com/longhorn/upgrade-responder/blob/6377af29b46a091d54847e68a87736f37abb50d4/upgraderesponder/service.go#L296

Reply:
OK we can leave the error. For now let's just return what user set in the response.

Reply:
Thank you. Fixed